low-level/sensors/impl/generic/raid.py

```diff
@@ -132,6 +140,40 @@ def initialize(self, conf_reader, msgQlist, product):
                                 self.SYSTEM_INFORMATION, COMMON_CONFIGS.get(self.SYSTEM_INFORMATION).get(self.RACK_ID), '001')
         self._node_id = self._conf_reader._get_value_with_default(
                                 self.SYSTEM_INFORMATION, COMMON_CONFIGS.get(self.SYSTEM_INFORMATION).get(self.NODE_ID), '001')
+
+        cache_dir_path = os.path.join(DATA_PATH, self.CACHE_DIR_NAME)
+        self.RAID_SENSOR_DATA_PATH = os.path.join(cache_dir_path,
+                                        f'RAID_SENSOR_DATA_{self._node_id}')
+        # Get the stored previous alert info
+        self.persistent_raid_data = {}
+        if os.path.isfile(self.RAID_SENSOR_DATA_PATH):
+            self.persistent_raid_data = store.get(self.RAID_SENSOR_DATA_PATH)
+        if self.persistent_raid_data:
+            self._RAID_status_contents = self.persistent_raid_data['_RAID_status_contents']
+            self._RAID_status = self.persistent_raid_data['_RAID_status']
+            self._faulty_drive_list = self.persistent_raid_data['_faulty_drive_list']
+            self._faulty_device_list = self.persistent_raid_data['_faulty_device_list']
+            self._drives = self.persistent_raid_data['_drives']
+            self._total_drives = self.persistent_raid_data['_total_drives']
+            self._devices = self.persistent_raid_data['_devices']
+            self._missing_drv = self.persistent_raid_data['_missing_drv']
+            self._prev_drive_dict = self.persistent_raid_data['_prev_drive_dict']
+            self.prev_alert_type = self.persistent_raid_data['prev_alert_type']
+        else:
+            self.persistent_raid_data = {
+                '_RAID_status_contents' : self._RAID_status_contents,
+                '_RAID_status' : self._RAID_status,
+                '_faulty_drive_list' : self._faulty_drive_list,
+                '_faulty_device_list' : self._faulty_device_list,
+                '_drives' : self._drives,
+                '_total_drives' : self._total_drives,
+                '_devices' : self._devices,
+                '_missing_drv' : self._missing_drv,
+                '_prev_drive_dict' : self._prev_drive_dict,
+                'prev_alert_type' : self.prev_alert_type,
+            }
+            store.put(self.persistent_raid_data, self.RAID_SENSOR_DATA_PATH)
+
         # Allow systemd to process all the drives so we can map device name to serial numbers
         #time.sleep(120)
         self.utility = Utility()
```

Comment:
I think this can be removed, since `sspl_constansts.py` already has a list
`DISABLE_FOR_VIRTUAL_SERVER = ("CPUFaultSensor", "RAIDIntegritySensor", "RAIDsensor", "MemFaultSensor", "NodeHWsensor")`
which prevents these sensors to run In virtual environemnt.

Reply:
no, these are self sensor measures, when someone adds up in config by mistake or without enough knowledge.
again, dont touch or change anything beyond decided scope of any task, it becomes difficult to understand, explain the reason and test scope increases unecessarily.

Reply:
okay..